packages/gatsby-cli/src/handlers/plugin-add-utils.ts

```diff
@@ -22,19 +23,41 @@ const addPluginToConfig = (
     key: string
   }
 ): string => {
-  const addPlugins = new BabelPluginAddPluginsToGatsbyConfig({
-    pluginOrThemeName: name,
-    options,
-    shouldAdd: true,
-    key,
-  })
+  let code
 
-  // @ts-ignore - fix me
-  const { code } = transform(src, {
-    // @ts-ignore - fix me
-    plugins: [addPlugins.plugin],
-    configFile: false,
-  })
+  try {
+    const transformOptions: TransformOptions = {
+      plugins: [
+        [
+          BabelPluginAddPluginsToGatsbyConfig,
+          {
+            pluginOrThemeName: name,
+            options,
+            key,
+          },
+        ],
+      ],
+      filename: srcPath,
+      configFile: false,
+    }
+
+    // Use the Babel TS preset if we're operating on `gatsby-config.ts`
+    if (srcPath.endsWith(`ts`)) {
+      transformOptions.presets = [`@babel/preset-typescript`]
+    }
+
+    code = transform(src, transformOptions)?.code
+
+    // Add back stripped type import, do light formatting, remove added empty module export.
+    // Use semicolon since Babel does that anyway, and we might as well be consistent.
+    if (srcPath.endsWith(`ts`)) {
+      code = `import type { GatsbyConfig } from 'gatsby;'\n\n${code}`
+      code = code.replace(`export {};`, ``)
```

Comment:
I'm not sure why this extra export gets added when there's already a default export. My working theory is Babel or the Babel TS preset sees an ES module with a default export and *requires* that it have a non-default export.

I couldn't find a way to configure this in the Babel docs, and I didn't want to do something like `export { config as default };` just to make Babel happy. So the replace option seemed reasonable since this transform is *only* for the JS and TS minimal starters, and we've got snapshots in case the starter configs change.

Happy to adjust if there's a better solution

Reply:
Ok, then let's keep it as-is. Was just wondering

Reply:
I would have asked too, it does look suspicious 😄